### ISSUE TITLE
SSL Enforcer doesn't work

### ISSUE THREAD

**Issue by danielmcoelho** [middleman/middleman#2060]
## Expected behavior and actual behavior
I want to force middleman to go HTTPS because I have the SSL certificate, so I've found `rack-ssl-enforcer`. It works locally but when is deployed to Heroku the build fails – `asset_hash` complains.
I'm on a webserver, with other middleware working nicely (e.g. rack-reverse-proxy).
The error in heroku build:
```
bundler: failed to load command: middleman (/tmp/build_bd2a0de6fbefbe5499d24a9a14033ebf/mysite-39098e1809812ab2fc4d1af3d683365ebd5bd969/vendor/bundle/ruby/2.3.0/bin/middleman)
RuntimeError: images/people/vladimir.jpg should be in the sitemap!
  /tmp/build_bd2a0de6fbefbe5499d24a9a14033ebf/mysite-39098e1809812ab2fc4d1af3d683365ebd5bd969/vendor/bundle/ruby/2.3.0/gems/middleman-core-3.4.0/lib/middleman-more/extensions/asset_hash.rb:48:in 'manipulate_single_resource'. 
```

My config.rb:
```
require 'rack/ssl-enforcer'
use Rack::SslEnforcer
...
configure :build do
  ignore %r{(?:(?:\w*/)?__)?showcase/.*}
  activate :minify_css
  activate :minify_javascript
  activate :gzip
  activate :minify_html
  activate :asset_hash
end
```
## Steps to reproduce the problem (from a clean middleman installation)
Install `rack-ssl-enforcer` and add `use Rack::SslEnforcer` to `config.rb`

## Additional information
I can't find in the documentation but maybe middleman has an option to enforce SSL certificate – I'm generating it with `letsencrypt`?

- Ruby version: 5.0
- Middleman version: 3.4.0
- OS version: Heroku is the problem, not my mac MacOS Sierra 10.12.2

**Comment by danielmcoelho** [comment]
TL;DR: How to force SSL certificate with Middleman? 😄

**Comment by geewiz** [comment]
By using Rack to serve your pages, `Rack::SSL` will do the trick.

Add to your `config.ru`:

```
# Force SSL
unless ENV["DEV_ENV"]
  use Rack::SSL
end
```
